Sources/LanguageServerProtocol/SupportTypes/LSPAny.swift

```diff
@@ -121,3 +121,52 @@ public protocol LSPAnyCodable {
   init?(fromLSPDictionary dictionary: [String: LSPAny])
   func encodeToLSPAny() -> LSPAny
 }
+
+extension Optional: LSPAnyCodable where Wrapped: LSPAnyCodable {
+  public init?(fromLSPAny value: LSPAny) {
+    if case .null = value {
+      self = .none
+      return
+    }
+    guard case .dictionary(let dict) = value else {
+      return nil
+    }
+    guard let wrapped = Wrapped.init(fromLSPDictionary: dict) else {
+      return nil
+    }
+    self = .some(wrapped)
+  }
+
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    return nil
+  }
+
+  public func encodeToLSPAny() -> LSPAny {
+    guard let wrapped = self else { return .null }
+    return wrapped.encodeToLSPAny()
```

Comment:
Do we actually use the `null` encoding anywhere, or is this just for completeness?  If possible I strongly prefer skipping nil values instead of encoding them as `null` since that seems to play nicer with other tools and saves space.

Reply:
Swapped over the place that used it